internal/gencli/gencli.go

```diff
@@ -441,6 +452,49 @@ func (g *gcli) addImport(cmd *Command, t pbinfo.ProtoType) (*pbinfo.ImportSpec,
 	return pkg, nil
 }
 
+func (g *gcli) addComments(f *descriptor.FileDescriptorProto, loc *descriptor.SourceCodeInfo_Location) {
+	var key proto.Message
+	p := loc.Path
+
+	switch {
+	// Service comment
+	case len(p) == 2 && p[0] == 6:
+		key = f.Service[p[1]]
+
+	// Method comment
+	case len(p) == 4 && p[0] == 6 && p[2] == 2:
+		key = f.Service[p[1]].Method[p[3]]
+
+	// Message comment
+	case len(p) == 2 && p[0] == 4:
+		key = f.MessageType[p[1]]
+
+	// Field comment
+	case len(p) == 4 && p[0] == 4 && p[2] == 2:
+		key = f.MessageType[p[1]].Field[p[3]]
+
+	// Extension comment
+	case len(p) == 2 && p[0] == 7:
+		key = f.Extension[p[1]]
+
+	// Enum comment
+	case len(p) == 2 && p[0] == 5:
+		key = f.EnumType[p[1]]
+
+	// Enum Value comment
+	case len(p) == 4 && p[0] == 5 && p[2] == 2:
+		key = f.EnumType[p[1]].Value[p[3]]
+
+	// TODO(ndietz): this is an incomplete mapping of comments
+	default:
+		return
+	}
+
+	if _, ok := g.comments[key]; !ok {
```

Comment:
Is this check necessary? I think we're in some weird place if we can reach one element two different ways.